lib/qio_field.cpp

```diff
@@ -28,10 +28,12 @@ std::ostream &operator<<(std::ostream &out, const QIO_Layout &layout)
   return out;
 }
 
+static int vlen;
```

Comment:
Not a QIO expoert but why have that globaly and set it in some funtctions as `vlen=len` but in others only use it? Can't we always pass that as argument where needed?

Reply:
We can't pass it in as an argument, since it it QIO that calls some of these functions, and we have no way to pass a parameter through QIO to the called function except through this file-local variable.

Reply:
I note: the former way I dealt with this was through templating the final functions, which works, but limits the possible `vlen` values to whatever is pre-instantiated.  This new approach removes that constraint.